koku/masu/celery/tasks.py

```diff
@@ -312,3 +315,36 @@ def vacuum_schemas():
     for schema_name in schema_names:
         LOG.info("Scheduling VACUUM task for %s", schema_name)
         vacuum_schema.delay(schema_name)
+
+
+@app.task(name="masu.celery.tasks.clean_volume", queue_name="clean_volume")
+def clean_volume():
+    """Clean up the volume in the worker pod."""
+    LOG.info("Cleaning up the volume at %s " % Config.PVC_DIR)
+    # get the billing months to use below
+    months = DateAccessor().get_billing_months(Config.INITIAL_INGEST_NUM_MONTHS)
```

Comment:
So, this would only ever clean out this month and the previous month right? So, if I had something in my volume from December it wouldn't be touched?

Reply:
It calculates the months that it uses to query the db by looking at the `INITIAL_INGEST_NUM_MONTHS` variable (which defaults to 2 - so unless you change that variable - the months to query for the latest seen manifest would be January and February). However the manifest ids that are returned from the query that uses the months are the ones that we don't want to delete - so on the contrary, if you had something in your volume from December, it would be deleted. The only way that files will not be deleted is if they are associated with the `assembly_ids` of the manifests that were most recently seen and have not been processed (which are filtered by month).

Reply:
Ohhh. Gotcha nah this perfect, thanks for the breakdown. Just wanted to make sure I understood what was happening.

Reply:
No worries! It's always hard to follow the code when you haven't been heads down on the issue! Thanks for your comments/questions 😄